src/shared/contexts/csv.worker.ts

```diff
@@ -0,0 +1,358 @@
+    []
+  )
+
+  const datatypeByColumnName = datatypeRow
+    .slice(1)
+    .reduce((acc, val, i) => ({...acc, [headerRow[i + 1]]: val}), {})
+
+  const defaultByColumnName = defaultRow
+    .slice(1)
+    .reduce((acc, val, i) => ({...acc, [headerRow[i + 1]]: val}), {})
+
+  return {groupKey, datatypeByColumnName, defaultByColumnName}
+}
+
+const TO_COLUMN_TYPE: {[fluxDatatype: string]: ColumnType} = {
+  boolean: 'boolean',
+  unsignedLong: 'number',
+  long: 'number',
+  double: 'number',
+  string: 'string',
+  'dateTime:RFC3339': 'time',
+}
+
+const parseValue = (value: string | undefined, columnType: ColumnType): any => {
+  if (value === undefined) {
+    return undefined
+  }
+
+  if (value === 'null') {
+    return null
+  }
+
+  if (value === 'NaN') {
+    return NaN
+  }
+
+  if (columnType === 'boolean' && value === 'true') {
+    return true
+  }
+
+  if (columnType === 'boolean' && value === 'false') {
+    return false
+  }
+
+  if (columnType === 'string') {
+    return value
+  }
+
+  if (columnType === 'time') {
+    return Date.parse(value.trim())
+  }
+
+  if (columnType === 'number' && value === '') {
+    return null
+  }
+
+  if (columnType === 'number') {
+    return Number(value)
+  }
+
+  return null
+}
+
+/*
+  Each column in a parsed `Table` can only have a single type, but because we
+  combine columns from multiple Flux tables into a single table, we may
+  encounter conflicting types for a given column during parsing.
+  To avoid this issue, we seperate the concept of the column _key_ and column
+  _name_ in the `Table` object, where each key is unique but each name is not
+  necessarily unique. We name the keys something like "foo (int)", where "foo"
+  is the name and "int" is the type.
+  But since type conflicts are rare and the public API requires referencing
+  columns by key, we want to avoid unwieldy keys whenever possible. So the last
+  stage of parsing is to rename all column keys from the `$NAME ($TYPE)` format
+  to just `$NAME` if we can do so safely. That is what this function does.
+*/
+const resolveNames = (
+  columns: Columns,
+  fluxGroupKeyUnion: Set<string>
+): void => {
+  const colNameCounts = Object.values(columns)
+    .map(col => col.name)
+    .reduce((acc, name) => ({...acc, [name]: (acc[name] || 0) + 1}), {})
+
+  const uniqueColNames = Object.entries(colNameCounts)
+    .filter(([_, count]) => count === 1)
+    .map(([name]) => name)
+
+  for (const uniqueName of uniqueColNames) {
+    const [columnKey, column] = Object.entries(columns).find(
+      ([_, col]) => col.name === uniqueName
+    )
+
+    columns[uniqueName] = column
+
+    delete columns[columnKey]
+
+    if (fluxGroupKeyUnion.has(columnKey)) {
+      fluxGroupKeyUnion.delete(columnKey)
+      fluxGroupKeyUnion.add(uniqueName)
+    }
+  }
+}
+
+const ctx: Worker = self as any
+
+ctx.addEventListener('message', msg => {
+  const [idx, csv] = msg.data
+  ctx.postMessage([idx, fromFlux(csv)])
+})
```

Comment:
so what I'm gather from this file is that the webworker can't make use of existing files or functions, or dependencies right? That's the reason for the pasta? Nothing new here right?

Reply:
nothing new, except we can't have anything more than a plain old javascript object being passed back and forth (had to remove simpeTable). we couldn't import a giraffe function directly because of how the exports are exposed in the npm module.